icu4j/main/classes/core/src/com/ibm/icu/impl/units/MeasureUnitImpl.java

```diff
@@ -752,8 +767,26 @@ public MeasureUnitImplComparator(ConversionRates conversionRates) {
 
         @Override
         public int compare(MeasureUnitImpl o1, MeasureUnitImpl o2) {
```

Comment:
For the future, please split orthogonal changes into their own PRs (eg, one for the ComplexUnitConverter fix and one for the compare fix).

Reply:
done and the pull request for the comparison has been submitted